packages/engine-server/src/workspace/utils.ts

```diff
@@ -68,14 +82,22 @@ export class WorkspaceUtils {
     return configPath;
   }
 
-  static findWSRootInWorkspaceFolders(
+  static async findWSRootInWorkspaceFolders(
     workspaceFolders: readonly WorkspaceFolderCode[]
-  ): WorkspaceFolderCode | undefined {
-    const dendronWorkspaceFolders =
-      workspaceFolders.filter((ent) => {
-        return fs.pathExistsSync(path.join(ent.uri.fsPath, "dendron.yml"));
-      }) || [];
-    return dendronWorkspaceFolders[0];
+  ): Promise<string | undefined> {
+    const folders = uniqueOutermostFolders(
+      workspaceFolders.map((folder) => folder.uri.fsPath)
+    );
+    const dendronWorkspaceFolders = await Promise.all(
+      folders.map((folder) =>
+        findDownTo({
+          base: folder,
+          fname: CONSTANTS.DENDRON_CONFIG_FILE,
+          returnDirPath: true,
+        })
+      )
+    );
+    return dendronWorkspaceFolders.filter(isNotUndefined)[0];
```

Comment:
instead of returning just the first entry, can we return all of them? in the short term, this lets us warn the user that we detected multiple and choose one. in the long term, we can eventually see if it makes sense to support this use case

Reply:
Revised this code to return all of them. Also added some code to Native Workspace initialization to prompt user if there are multiple roots. This prompt should be pretty rare because the only situation this would happen is if:
1. User has multiple workspace folders configured
2. There are 2 workspace folders, not subfolders of each other, that both contain `dendron.yml`
I tested this with a workspace I created with this exact setup:

![](https://i.imgur.com/RXm2O09.png)
![](https://i.imgur.com/ccckuO9.png)